_docs-2/administration/compaction.md

```diff
@@ -62,6 +68,151 @@ For more information see the javadoc for {% jlink org.apache.accumulo.core.spi.c
+
+When a Compactor is free to perform work, it asks the CompactionCoordinator for the next compaction job. The CompactionCoordinator contacts the next TabletServer that has the highest priority for the Compactor's queue. The TabletServer returns the information necessary for the compaction to occur to the CompactionCoordinator, which is passed on to the Compactor. The Compaction Coordinator maintains an in-memory list of running compactions and also inserts an entry into the metadata table for the tablet to denote that an external compaction is running. When the Compactor has finished the compaction, it notifies the CompactionCoordinator which inserts an entry into the metadata table to denote that the external compaction completed and it attempts to notify the TabletServer. If successful, the TabletServer commits the major compaction. If the TabletServer is down, or the Tablet has become hosted on a different TabletServer, then the CompactionCoordinator will fail to notify the TabletServer, but the metadata table entries will remain. The major compaction will be committed in the future by the TabletServer hosting the Tablet.
+
+### External Compaction in Action
+
+Below are some examples of log entries and metadata table entries for external compactions. First, here are some metadata entries for table `2` . You can see that there are three files of different sizes (file size and number of entries are stored in the value portion of the metadata table rows with the "file" column qualifier).
+
+```
+2< file:hdfs://localhost:8020/accumulo/tables/2/default_tablet/A0000047.rf []	12330,99000
+2< file:hdfs://localhost:8020/accumulo/tables/2/default_tablet/F0000048.rf []	1196,1000
+2< file:hdfs://localhost:8020/accumulo/tables/2/default_tablet/F000004j.rf []	1302,1000
+2< last:10000bf4e0a0004 []	localhost:9997
+2< loc:10000bf4e0a0004 []	localhost:9997
+2< srv:compact []	111
+2< srv:dir []	default_tablet
+2< srv:flush []	113
+2< srv:lock []	tservers/localhost:9997/zlock#1950397a-b2ca-4685-b70b-67ae3cd578b9#0000000000$10000bf4e0a0004
+2< srv:time []	M1618325648093
+2< ~tab:~pr []	\x00
+```
+
+Below are excerpts from the TabletServer, CompactionCoordinator, Compactor logs and metadata table. I have merged the logs in time order to make it easier to see what is happening.
+
+In the logs below the Compactor requested a compaction job from the Coordinator with an ExternalCompactionId of `de6afc1d-64ae-4abf-8bce-02ec0a79aa6c`. The Coordinator knew that TabletServer `localhost:9997` had a Tablet that needed compacting and contacted it to get the details. The CompactionManager, a component
+running in the TabletServer, returned the information to the Coordinator. The Coordinator then updates the metadata table (below the logs) for the external compaction and returns the information to the Compactor:
+
```

Comment:
Should this example show starting the coordinator and compactor?

Reply:
I think it should. Maybe a full example within `accumulo-examples` could be helpful as well in the future.

Reply:
Resolved in 5314ec2bc0215fe75f7f7b866fe3dd26e7cacb87